apps/trivium/src/trivium/trivium_byte.rs

```diff
@@ -0,0 +1,258 @@
+            || {
+                rayon::join(
+                    || rayon::join(|| a4 ^ a5, || b4 ^ b5),
+                    || rayon::join(|| c4 ^ c5, || a1 & a2),
+                )
+            },
+            || rayon::join(|| b1 & b2, || c1 & c2),
+        );
+
+        let (temp_a_2, temp_b_2, temp_c_2) = (temp_a.clone(), temp_b.clone(), temp_c.clone());
+
+        let ((o, a), (b, c)) = rayon::join(
+            || {
+                rayon::join(
+                    || (temp_a_2 ^ temp_b_2) ^ temp_c_2,
+                    || temp_c ^ ((c_and) ^ a3),
+                )
+            },
+            || rayon::join(|| temp_a ^ (a_and ^ b3), || temp_b ^ (b_and ^ c3)),
+        );
+
+        [o, a, b, c]
+    }
+
+    /// This calls `get_output_and_values` in parallel 8 times, and stores all results in a Vec.
+    fn get_64_output_and_values(&self) -> Vec<[T; 4]> {
+        let copies = vec![
+            (
+                self.a_byte.clone(),
+                self.b_byte.clone(),
+                self.c_byte.clone()
+            );
+            8
+        ];
+        copies
+            .into_par_iter()
+            .enumerate()
+            .map(|(i, (a, b, c))| Self::get_output_and_values(a, b, c, i))
+            .collect()
+    }
+
+    /// Computes 64 turns of the stream, outputting the 64 bits (in 8 bytes) all at once in a
+    /// Vec (first value is oldest, last is newest)
+    pub fn next_64(&mut self) -> Vec<T> {
+        match &self.fhe_key {
+            Some(sk) => {
+                rayon::broadcast(|_| set_server_key(sk.clone()));
+            }
+            None => (),
+        }
+        let values = self.get_64_output_and_values();
+        match &self.fhe_key {
+            Some(_) => {
+                rayon::broadcast(|_| unset_server_key());
+            }
+            None => (),
+        }
+
+        let mut bytes = Vec::<T>::with_capacity(8);
+        for [o, a, b, c] in values {
+            self.a_byte.push(a);
+            self.b_byte.push(b);
+            self.c_byte.push(c);
+            bytes.push(o);
+        }
+
+        bytes
+    }
+
+    /// Reconstructs a bunch of 5 bytes in a parallel fashion.
+    fn get_bytes<const N: usize>(
+        reg: StaticBitDeque<N, T>,
+        offsets: [usize; 5],
+    ) -> (T, T, T, T, T) {
+        let [reg1, reg2, reg3, reg4, reg5] = [
+            reg.clone(),
+            reg.clone(),
+            reg.clone(),
+            reg.clone(),
+            reg.clone(),
+        ];
+        let (ret5, ((ret1, ret2), (ret3, ret4))) = rayon::join(
+            move || reg5.byte(offsets[4]),
+            || {
+                rayon::join(
+                    || rayon::join(move || reg1.byte(offsets[0]), move || reg2.byte(offsets[1])),
+                    || rayon::join(move || reg3.byte(offsets[2]), move || reg4.byte(offsets[3])),
+                )
+            },
+        );
+        (ret1, ret2, ret3, ret4, ret5)
```

Comment:
par_iter_mut + enumerate and a match on the index maybe to fill with the correct offset ?

I'm not a fan of nested rayon join